login-Example/login-Example/ViewController.swift

```diff
@@ -113,4 +115,90 @@ class ViewController: UIViewController, InfomaniakLoginDelegate, DeleteAccountDe
         })
         present(alertController, animated: true, completion: nil)
     }
+
+    @IBAction func refreshTokenConvert(_ sender: Any) {
+        if #available(iOS 13.0, *) {
+            SimpleResolver.sharedResolver.removeAll()
+            // Init with non infinite refresh token
+            SimpleResolver.sharedResolver.store(factory: Factory(type: InfomaniakLoginable.self) { _, _ in
+                let clientId = "9473D73C-C20F-4971-9E10-D957C563FA68"
+                let redirectUri = "com.infomaniak.drive://oauth2redirect"
+                let login = InfomaniakLogin(config: .init(clientId: clientId, redirectURI: redirectUri, accessType: .offline))
+                return login
+            })
+            SimpleResolver.sharedResolver.store(factory: Factory(type: InfomaniakTokenable.self) { _, resolver in
+                return try resolver.resolve(type: InfomaniakLoginable.self,
+                                            forCustomTypeIdentifier: nil,
+                                            factoryParameters: nil,
+                                            resolver: resolver)
+            })
+
+            @InjectService var loginService: InfomaniakLoginable
+            @InjectService var tokenService: InfomaniakTokenable
+
+            loginService.asWebAuthenticationLoginFrom(anchor: .init(),
+                                                      useEphemeralSession: false,
+                                                      hideCreateAccountButton: true) { result in
+                switch result {
+                case .success(let success):
+                    self.tokenService.getApiTokenUsing(code: success.code, codeVerifier: success.verifier) { token, error in
+                        var title: String?
+                        var description: String?
+
+                        if let token = token {
+                            title = "Login completed"
+                            description = "UserId: \(token.userId)\nToken: \(token.accessToken)\nExpires in: \(token.expiresIn ?? -1)"
+                            self.testSwapRefreshToken(apiToken: token)
+                        } else if let error = error {
+                            title = "Login error"
+                            description = error.localizedDescription
+                        }
+                        print("refreshTokenConvert \(title ?? "")\n\(description ?? "")")
+                    }
+                case .failure(let failure):
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+                        self.showAlert(title: "Error", message: failure.localizedDescription)
+                    }
+                }
+            }
+        }
+    }
+    
+    func testSwapRefreshToken(apiToken: ApiToken) {
```

Comment:
Yeah this should be made an actual test. Maybe add a todo for autologin ?

Reply:
I also would you like to test it but it is hard to do since we need to go through a webview to get a token.
As discussed faking network calls would be a great step forward.